### PR TITLE
feat: jwt signing

### PR DESCRIPTION
**Pull request by thepiwo**
Currently based on the latest 13.2.2 release, thus the conflicts to try it easily with the current superhero wallet implementation.

related superhero wallet integration https://github.com/superhero-com/superhero-wallet/pull/2875

### CODE REVIEW

File: examples/browser/aepp/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "devDependencies": {
     "@vue/cli-plugin-babel": "~5.0.8",
     "@vue/cli-service": "~5.0.8",
+    "postcss": "^8.4.37",
```
Comment (thepiwo) [inline examples/browser/aepp/package.json:21]: seems to be needed to start the example

**Comment by thepiwo** [comment]
@davidyuk any feedback from your side?

**Comment by davidyuk** [comment]
The motivation is unclear. Have you considered using [typed data](https://github.com/aeternity/aepp-sdk-js/blob/develop/docs/guides/typed-data.md) instead of JWT? Or it is not suitable for your use case?

**Comment by thepiwo** [comment]
JWT is nice as its a widely used standard and supported by our curve, so we could expose it (and e.g. use it in superhero product login)

**Review by davidyuk — COMMENTED** [review]
Not sure about motivation, seems that JWTs signed by a centralized provider are more useful/compatible that self-signed JWTs.

File: src/utils/crypto.ts [hunk 2]
```diff
@@ -142,6 +143,18 @@ export function sign(data: string | Uint8Array, privateKey: string | Uint8Array)
   return nacl.sign.detached(Buffer.from(data), Buffer.from(privateKey));
 }
 
+export async function signJWT(
+  message: object,
+  expireAt: number,
+  privateKey: Uint8Array,
+): Promise<string> {
+  const header = { alg: 'EdDSA', typ: 'JWT' };
+  const payload = { ...message, exp: expireAt };
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/utils/crypto.ts:152]: agree!

File: src/utils/crypto.ts [hunk 3]
```diff
@@ -142,6 +143,18 @@ export function sign(data: string | Uint8Array, privateKey: string | Uint8Array)
...
+  const body = `${base64url.encode(JSON.stringify(header))}.${base64url.encode(JSON.stringify(payload))}`;
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/utils/crypto.ts:153]: nice, didn't know

File: src/utils/crypto.ts [hunk 4]
```diff
@@ -142,6 +143,18 @@ export function sign(data: string | Uint8Array, privateKey: string | Uint8Array)
...
+  const signature = sign(body, privateKey);
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/utils/crypto.ts:170]: if we add `sub_jwk` in the payload the risk should be even smaller

File: src/account/Memory.ts [hunk 5]
```diff
@@ -77,6 +77,19 @@ export default class AccountMemory extends AccountBase {
...
   }
 
+  override async signMessageJWT(
+    message: object,
+    options?: { expireAt?: number },
+  ): Promise<string> {
+    const secretKey = secretKeys.get(this);
+    if (secretKey == null) throw new UnexpectedTsError();
+    const expireAt = options?.expireAt === undefined
+      ? new Date().getTime() + 30 * 60 * 1000 // default to 30min
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/account/Memory.ts]: do you think its ok to keep the default expiry there?

File: src/account/Rpc.ts [hunk 6]
```diff
@@ -53,6 +53,14 @@ export default class AccountRpc extends AccountBase {
     return Buffer.from(signature, 'hex');
   }
 
+  override async signMessageJWT(message: object, options?: {
+    expireAt?: number;
+  }): Promise<string> {
+    const { signature } = await this._rpcClient
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/account/Rpc.ts]: renamed it

File: src/account/Memory.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (davidyuk) [reply src/account/Memory.ts]: Not really, there are different account types and I'm not sure each of them should define it. But here is not much logic to extract to "generate jwt" function. We can do nothing here and allow the wallet to handle jwt parameters (like not allowing signing a too general token).

[the remark this answers is not in the document]
Comment (thepiwo) [reply src/account/Memory.ts]: I removed it :)

**Pull request closed without merging**